engine/consensus/matching/engine.go

```diff
@@ -620,6 +650,10 @@ func (e *Engine) checkingSealing() error {
 		Int("sealable_incorporated_results", len(sealedBlockIDs)).
 		Int64("duration_ms", time.Since(startTime).Milliseconds()).
 		Bool("mempool_has_next_seal", mempoolHasNextSeal).
```

Comment:
⚠️ adding this as a comment to this PR:
I think there is an edge case, where `mempoolHasNextSeal` is true but sealing can't progress:
* we only check whether the result for the block is sealable
* but we don't check whether the parent result is actually the one which was sealed.

Example:
* Lets say we have blocks `A <- B`
   - block `A` was sealed with result `r_A`
   - however, for block `A`, we might also know a different result `r_A'` which was _not_ sealed
   - for block `B`, we have two execution results:
      -  `r_B` (with parent `r_A`), which has _not_ acquired enough approvals 
      -  `r_B'` (with parent `r_A'`), which has acquired enough approvals 
    - We cannot seal result `r_B'` because its parent conflicts with the sealed one

I think your logic will log `mempoolHasNextSeal = True`, even though this is not the case.

Reply:
True. 

Although you are talking about the case where there is an execution fork, in which case, the node will crash, and we won't see this log. 

So if we ever see this log, it means, there was no execution fork, but if sealing still halts, this log could be helpful for us to debug.